From ba9ac907d3f359fe0f9321e026aaadd6dfe3f6a5 Mon Sep 17 00:00:00 2001
From: Ludovic Beliveau <ludovic.beliveau@windriver.com>
Date: Mon, 21 Mar 2016 18:12:32 -0400
Subject: [PATCH] STX: System Logging: set group read permission on log files

Commit carried over from pre-CentOS

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
[ Fixed pointers while porting patch to libvirt v8.0.0-1 ]
Signed-off-by: Daniel Caires <danielmarques.caires@windriver.com>
[ Ported to libvirt v11.3.0-3+deb13u2 ]
Signed-off-by: Murillo Arantes <murillo.arantes@windriver.com>
---
 src/hypervisor/domain_logcontext.c | 2 +-
 src/qemu/qemu_domain.c             | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/hypervisor/domain_logcontext.c b/src/hypervisor/domain_logcontext.c
index 41d1bbdf6..a8a063f54 100644
--- a/src/hypervisor/domain_logcontext.c
+++ b/src/hypervisor/domain_logcontext.c
@@ -111,7 +111,7 @@ domainLogContextNew(bool stdioLogD,
         if (ctxt->writefd < 0)
             goto error;
     } else {
-        if ((ctxt->writefd = open(ctxt->path, O_WRONLY | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR)) < 0) {
+        if ((ctxt->writefd = open(ctxt->path, O_WRONLY | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR | S_IRGRP)) < 0) {
             virReportSystemError(errno, _("failed to create logfile %1$s"),
                                  ctxt->path);
             goto error;
diff --git a/src/qemu/qemu_domain.c b/src/qemu/qemu_domain.c
index 52da23434..ac8aabec6 100644
--- a/src/qemu/qemu_domain.c
+++ b/src/qemu/qemu_domain.c
@@ -5712,7 +5712,7 @@ qemuDomainLogAppendMessage(virQEMUDriver *driver,
                                              vm->def->name, path, message, 0) < 0)
             goto cleanup;
     } else {
-        if ((writefd = open(path, O_WRONLY | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR)) < 0) {
+        if ((writefd = open(path, O_WRONLY | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR | S_IRGRP)) < 0) {
             virReportSystemError(errno, _("failed to create logfile %1$s"),
                                  path);
             goto cleanup;
@@ -10381,7 +10381,7 @@ virQEMUFileOpenAs(uid_t fallback_uid,
             goto error;
         }
     } else {
-        if ((fd = virFileOpenAs(path, oflags, S_IRUSR | S_IWUSR, uid, gid,
+        if ((fd = virFileOpenAs(path, oflags, S_IRUSR | S_IWUSR | S_IRGRP, uid, gid,
                                 vfoflags | VIR_FILE_OPEN_NOFORK)) < 0) {
             /* If we failed as root, and the error was permission-denied
                (EACCES or EPERM), assume it's on a network-connected share
-- 
2.34.1

